components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/Types.kt

```diff
@@ -192,15 +194,23 @@ internal fun HistoryMetadata.into(): mozilla.appservices.places.HistoryMetadata
     )
 }
 
-internal fun HistoryMetadataObservation.into(): mozilla.appservices.places.HistoryMetadataObservation {
+internal fun HistoryMetadataObservation.into(
+    key: HistoryMetadataKey
+): mozilla.appservices.places.uniffi.HistoryMetadataObservation {
```

Comment:
Interesting. Looks like there are now specific methods to record those observations we could use instead of converting the types? https://github.com/mozilla/application-services/blob/3376ccc2e06736a2811fe0ace2af131e98f2c78e/components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt#L672

I guess not reason to change that and we can still call the generic `noteHistoryMetadataObservation`?